File: Sources/AndroidBackend/ViewOnContextClickListener.swift
import AndroidKit
import SwiftJava

@JavaClass(
    "dev.swiftcrossui.androidbackend.ViewOnContextClickListener",
    implements: AndroidView.View.OnContextClickListener.self
)
class ViewOnContextClickListener: JavaObject {
    @JavaMethod
    @_nonoverride convenience init(action: SwiftAction?, environment: JNIEnvironment? = nil)
}

extension ViewOnContextClickListener {
    convenience init(action: @escaping () -> (), environment: JNIEnvironment? = nil) {
        let object = SwiftAction(environment: environment, action: action)
        self.init(action: object, environment: environment)
    }
}

File: Sources/AndroidBackend/Kotlin/ViewOnContextClickListener.kt
package dev.swiftcrossui.androidbackend

import android.view.View

/// The mouse/stylus half of a context menu. `ViewOnLongClickListener` is the
/// touch half; Android keeps them on two separate listener slots, which is why
/// there are two classes rather than one.
class ViewOnContextClickListener(private val action: SwiftAction) : View.OnContextClickListener {
    override fun onContextClick(view: View): Boolean {
        action.call()
        return true
    }
}
